keep-ui/app/error.tsx

```diff
@@ -1,36 +1,51 @@
 "use client";
 import Image from "next/image";
 import "./error.css";
-
-// Custom Error Component
+import {useEffect} from "react";
+import {Button, Text} from "@tremor/react";
 export default function ErrorComponent({
-  errorMessage,
-  url,
+  error,
+  reset,
 }: {
-  errorMessage: String;
-  url: String;
+  error: KeepApiError
+  reset: () => void
 }) {
+  useEffect(() => {
+    console.error(error)
+  }, [error])
+
   return (
     <div className="error-container">
-      <div className="error-message">{errorMessage}</div>
-      {url && (
+      <div className="error-message">{error.toString()}</div>
         <div className="error-url">
-          Failed to query {url}, is Keep API up?
+          {error.proposedResolution}
         </div>
-      )}
       <div className="error-image">
         <Image src="/keep.svg" alt="Keep" width={150} height={150} />
       </div>
+      <Button
+          onClick={() => {
+            console.log("Refreshing...")
+            window.location.reload();
+          }}
+          color="orange"
+          variant="secondary"
+          className="mt-4 border border-orange-500 text-orange-500">
+        <Text>Try Again!</Text>
+      </Button>
     </div>
-  );
+  )
 }
 
+// Custom Error Class
 export class KeepApiError extends Error {
   url: string;
+  proposedResolution: string;
 
-  constructor(message: string, url: string) {
+  constructor(message: string, url: string, proposedResolution: string) {
```

Comment:
What if it's some other exception, not "KeepApiError"?

Reply:
Handled that case.

Reply:
Can you show how it looks?

Reply:
@talboren 
![image](https://github.com/user-attachments/assets/aa43db08-dc5c-485b-9553-188e0383dff7)